src/pattern-matching/exercise.rs

```diff
@@ -44,11 +44,11 @@ fn eval(e: Expression) -> Result<i64, String> {
         Expression::Op { op, left, right } => {
             let left = match eval(*left) {
                 Ok(v) => v,
-                e @ Err(_) => return e,
+                Err(e) => return Err(e),
             };
             let right = match eval(*right) {
                 Ok(v) => v,
-                e @ Err(_) => return e,
+                Err(e) => return Err(e),
```

Comment:
I agree it's good to avoid introducing new syntax in a solution; this always surprises students. But do we have any other place where this syntax is introduced? I think it is worth mentioning.

Reply:
I don't think it's worth mentioning, honestly. It's a rarely-used syntax and something students can pretty easily look up if they run into it in the wild. At most maybe we could have a speaker note on the patterns page, but I wouldn't put it in the slides.

Reply:
A speaker note seems like the right place for something minor like this IMO.

Reply:
Added.